library/Zend/Console/Getopt.php

```diff
@@ -789,7 +789,7 @@ protected function _parseLongOption(&$argv)
     protected function _parseShortOptionCluster(&$argv)
     {
         $flagCluster = ltrim(array_shift($argv), '-');
-        foreach (str_split($flagCluster) as $flag) {
+        foreach ($flagCluster === '' ?  [''] : str_split($flagCluster) as $flag) {
```

Comment:
you probably want to fallback to empty array not array containing empty value?

also for code readability, create new variable before foreach so foreach could be simple.

Reply:
Thanks, updated code @glensc

Reply:
it's nice if you respond with commit hash in comments then the reviewer can check the change easily.

Reply:
Oh sorry about that  Mr @glensc . [Here is commit hash](https://github.com/Shardj/zf1-future/pull/275/commits/2822b923c64a920d02eac8f97e3089a2b2ce6855#diff-46a176382becc2973de69e45c568a16e53e5d42890b0cbbd74a621247a509c36)

Reply:
In [`2822b92` (#275)](https://github.com/Shardj/zf1-future/pull/275/commits/2822b923c64a920d02eac8f97e3089a2b2ce6855#diff-46a176382becc2973de69e45c568a16e53e5d42890b0cbbd74a621247a509c36) you only created separate variable, what about using empty array `[]` instead of array of empty string `['']`? if the input is empty there's no point to foreach the loop (and empty array does not run the loop).

Reply:
He restored the "old behaviour" of PHP <8.2 by putting an empty string into the array. Since PHP 8.2 it returns an empty array, which does not run the loop. Maybe the `foreach` should be run with an empty string, like before?

https://php.watch/versions/8.2/str_split-empty-string-empty-array

Reply:
Does the code do anything useful with the empty string? maybe "fix" the code too?

Reply:
Mr @glensc  i have detail problem here in section "Root Cause". https://github.com/Shardj/zf1-future/issues/276

Short desc: in case  $flagCluster = '';  //empty string
- PHP < 8.2 : str_split($flagCluster) === [""] => program work as expected (test case pass)
- PHP >= 8.2: str_split($flagCluster) === [] => program work as expected (test case fail)
[Code sample online here](https://onlinephp.io?s=s7EvyCjg5SpLLIpPKc0t0CguKYovLsjJLNFQV9fUtAYA&v=8.2.0RC4%2C8.1.12%2C8.0.25%2C7.4.33%2C7.3.33)

So my code to keep logic on PHP >= 8.2 same with PHP < 8.2 in case  $flagCluster = '';

Reply:
that's why I asked ["you probably want to"](https://github.com/Shardj/zf1-future/pull/275#discussion_r1022473548), since you now gave more info, seems to me that fixing relying on the bug causes test to fall, is better keep with the previous behavior.